tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -302,6 +302,37 @@ def dir_size():
     assert dir_size() == 0
 
 
+def test_large_manifests_passed_by_file(wandb_init, monkeypatch, mocker):
```

Comment:
why do you need a system test for this? could you get away with writing a unit test in go directly?

Reply:
i see, you implemented it the python code as well... 
i would suggest adding the unit test in go anyways, better for code quality and potential future changes

Reply:
1. we want to make sure both the python and the go cores operate the same way
2. half of the PR is creating the file and passing it into the core process

So I could write three unit tests (one for encoding and sending, one for handling in Python, one for handling in Go) and assume that they all work together, or I can write an integration test that is basically this one plus those three, but I admit I'm not enthused to write the unit tests for either the encode in the sender (because there aren't super easy injection points) or unit tests for receiving in Go (because that also doesn't have much of a test harness to work with)

Reply:
hmmm... i don't know we were able to write test for other pieces of the stream code quite easily with some of the testing packages we added!!!

Reply:
By lack of a test harness I really mean for instantiating and interacting with Artifacts* objects, so that's really on us, not a criticism of the SDK more broadly. I'll take a stab at it and see how straightforward I can make this test.